ui/src/cells/actions/thunks.ts

```diff
@@ -0,0 +1,162 @@
+import {setView} from 'src/dashboards/actions/views'
+import {notify} from 'src/shared/actions/notifications'
+import {setCells, setCell, removeCell} from 'src/cells/actions/creators'
+
+// Utils
+import {getClonedDashboardCell} from 'src/dashboards/utils/cellGetters'
+
+// Constants
+import * as copy from 'src/shared/copy/notifications'
+
+// Types
+import {
+  Dashboard,
+  NewView,
+  Cell,
+  GetState,
+  RemoteDataState,
+  NewCell,
+  DashboardEntities,
+  ResourceType,
+  CellEntities,
+} from 'src/types'
+
+// Utils
+import {getViewsForDashboard} from 'src/dashboards/selectors'
+import {getNewDashboardCell} from 'src/dashboards/utils/cellGetters'
+import {getByID} from 'src/resources/selectors'
+
+export const deleteCell = (dashboardID: string, cellID: string) => async (
+  dispatch,
+  getState: GetState
+): Promise<void> => {
+  try {
+    const views = getViewsForDashboard(getState(), dashboardID).filter(
+      view => view.cellID !== cellID
+    )
+
+    await Promise.all([
+      api.deleteDashboardsCell({dashboardID: dashboardID, cellID: cellID}),
+      dispatch(refreshDashboardVariableValues(dashboardID, views)),
+    ])
+
+    dispatch(removeCell({dashboardID, id: cellID}))
+    dispatch(notify(copy.cellDeleted()))
+  } catch (error) {
+    console.error(error)
+  }
+}
+
+export const createCellWithView = (
+  dashboardID: string,
+  view: NewView,
+  clonedCell?: Cell
+) => async (dispatch, getState: GetState): Promise<void> => {
+  const state = getState()
+
+  let dashboard = getByID<Dashboard>(
+    state,
+    ResourceType.Dashboards,
+    dashboardID
+  )
+
+  try {
+    if (!dashboard) {
+      const resp = await api.getDashboard({dashboardID})
+      if (resp.status !== 200) {
+        throw new Error(resp.data.message)
+      }
+
+      const {entities, result} = normalize<
+        Dashboard,
+        DashboardEntities,
+        string
+      >(resp.data, schemas.dashboard)
+
+      dashboard = entities.dashboards[result]
+    }
+
+    const cell: NewCell = getNewDashboardCell(state, dashboard, clonedCell)
+
+    // Create the cell
+    const cellResp = await api.postDashboardsCell({dashboardID, data: cell})
+
+    if (cellResp.status !== 201) {
+      throw new Error(cellResp.data.message)
+    }
+
+    const cellID = cellResp.data.id
+
+    // Create the view and associate it with the cell
+    const newView = await dashAPI.updateView(dashboardID, cellID, view)
+
+    const normCell = normalize<Cell, CellEntities, string>(
+      {...cellResp.data, dashboardID},
+      schemas.cell
+    )
+
+    // Refresh variables in use on dashboard
+    const views = [...getViewsForDashboard(state, dashboardID), newView]
+
+    await dispatch(refreshDashboardVariableValues(dashboardID, views))
+
+    dispatch(setView(cellID, newView, RemoteDataState.Done))
+    dispatch(setCell(cellID, RemoteDataState.Done, normCell))
+  } catch {
+    notify(copy.cellAddFailed())
```

Comment:
is this worthy of sending to honeybadger? maybe we should make a GA event for it if it's an error that's not a runtime error?

Reply:
Yea, I think there's larger work to be done for how we handle all http api errors.